File: euler39.py
import math


nums = [(a, b, math.sqrt(a**2 + b**2)) for a in range(1, 425) for b in range(1, a)]

def do_stuff(p):
    res = 0
    for a, b, c in nums:
        if a + b + c == p:
            res += 1
    return res

ans = 0
perim = 0
for x in range(1, 1000):
    tmp = do_stuff(x)
    if ans < tmp:
        ans = tmp
        perim = x

print perim
print ans

File: euler39.c
#include <math.h>
#include <stdio.h>

int do_stuff(double p) {
    int res = 0;
    double a, b, c = 0.0;
    for (a = 0.0; a < (p / 2); a++) {
	for (b = 0.0; b < a; b++) {
	    c = sqrt(pow(a,2.0) + pow(b,2.0));
	    if ((a + b + c) == p) {
		res++;
	    }
	}
    }

    return res;
}


int main () {
    int ans = 0, perim = 1, tmp = 0, x;
    for (x = 0; x <= 1000; x++) {
	tmp = do_stuff(x);
	if (ans < tmp) {
	    ans = tmp;
	    perim = x;
	}
    }
    printf("%d\n", ans);
    printf("%d\n", perim);

    return 0;
}
